cms/cms_toolbars.py

```diff
@@ -325,31 +395,8 @@ def populate(self):
         self.change_language_menu()
 
     def post_template_populate(self):
-        self.init_placeholders()
-        self.add_draft_live()
-        self.add_structure_mode()
-
-    # Buttons
-    def add_draft_live(self):
-        if self.page:
-            if self.toolbar.edit_mode_active and not self.title:
-                self.add_page_settings_button()
-
-            if user_can_change_page(self.request.user, page=self.page):
-                return self.add_draft_live_item()
-
-        elif self.placeholders:
-            return self.add_draft_live_item()
-
-        for sp in self.statics:
-            if sp.has_change_permission(self.request):
-                return self.add_draft_live_item()
-
-    def add_draft_live_item(self, template='cms/toolbar/items/live_draft.html', extra_context=None):
-        context = {'cms_toolbar': self.toolbar}
-        context.update(extra_context or {})
-        pos = len(self.toolbar.right_items)
-        self.toolbar.add_item(TemplateItem(template, extra_context=context, side=self.toolbar.RIGHT), position=pos)
+        if self.page and self.toolbar.edit_mode_active and not self.title:
```

Comment:
is this intentional? shouldn't it be `and self.title` instead of `and not self.title`?

Reply:
@czpython I basically kept the existing code as it is but when I checked it doesn't do anything and I think its because of the `and not self.title`. Is this function necessary as the Page settings item is added under Page menu?

Reply:
I've checked the code, add_page_settings_button has the following line:
`url = '%s?language=%s' % (admin_reverse('cms_page_change', args=[self.title.pk]), self.toolbar.request_language)`

Which would fail, as `self.title` would be None/EmptyPageContent. If anything, shouldn't this line be:
`url = '%s?language=%s' % (admin_reverse('cms_page_advanced', args=[self.page.pk]), self.toolbar.request_language)`?

Reply:
Unless I'm missing something (very possible), we don't need add_page_settings_button at all.
The page settings are added in line 562 and advanced settings in 568.
So remove this method and the call to it 🙈

Reply:
Just out of curiosity, is this a legacy bit that handled displaying page settings button for when you're viewing a page with no title for specified language?

Reply:
Yes and no, looks like it was adapted for 4.0.x changes but is not actually needed